app/models/queries/projects/filters/custom_field_filter.rb

```diff
@@ -30,184 +30,43 @@
 
 class Queries::Projects::Filters::CustomFieldFilter <
   Queries::Projects::Filters::ProjectFilter
-  attr_accessor :custom_field
 
-  validate :custom_field_valid
+  include Queries::Filters::Shared::CustomFieldFilter
 
-  def allowed_values
-    case custom_field.field_format
-    when 'bool'
-      [[I18n.t(:general_text_yes), CustomValue::BoolStrategy::DB_VALUE_TRUE],
-       [I18n.t(:general_text_no), CustomValue::BoolStrategy::DB_VALUE_FALSE]]
-    when 'version', 'list' # TODO: Why list?
-      # custom_field.possible_values_options(project)
-      custom_field.possible_values_options
-    when 'user'
-      custom_field.possible_values_options(:of_all_projects)
-    end
-  end
+  self.custom_field_class = ProjectCustomField
 
   def type
-    return nil unless custom_field
-
-    case custom_field.field_format
-    when 'int'
-      :integer
-    when 'float'
+    if custom_field && custom_field.field_format == 'float'
```

Comment:
This extra treatment for project custom fields of type float might not be necessary, as I believe, that the for work packages CFs of type float do not work as expected neither. So maybe we could solve that there too...